### PR TITLE
Invalidation

### PR DESCRIPTION
**Pull request by jeremyestein**
Invalidation fixes, plus smart comparison of pathology result facts.

### CODE REVIEW

File: src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java [hunk 1]
```diff
@@ -262,6 +265,61 @@ public int hashCode() {
         return result;
     }
 
+    /**
+     * Test equality of properties, optionally ignoring one or more by property type.
+     * @param other               the other fact to test property equality to
+     * @param attributesToIgnore  attributes to ignore when determining equality
+     * @return true iff all properties that you care about are equal, ignoring order of properties
```
Comment (stefpiatek) [inline src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java:272]:
```suggestion
     * @return true if all properties that you care about are equal, ignoring order of properties
```

File: src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java [hunk 2]
```diff
@@ -262,6 +265,61 @@ public int hashCode() {
...
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Test equality of facts, optionally ignoring one or more by properties by type.
+     * @param obj                 the other fact to test equality to
+     * @param attributesToIgnore  attributes to ignore when determining equality of properties
+     * @return true iff facts are equal, and all properties that you care about are equal, ignoring order of properties
```
Comment (stefpiatek) [inline src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java:297]:
```suggestion
     * @return true if facts are equal, and all properties that you care about are equal, ignoring order of properties
```

File: src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java (hunk 1 above)
Comment (jeremyestein) [reply src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java:272]: This is as intended - if and only if.

File: src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java (hunk 2 above)
Comment (jeremyestein) [reply src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java:297]: ditto

File: src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java (hunk 1 above)
Comment (stefpiatek) [reply src/main/java/uk/ac/ucl/rits/inform/informdb/Fact.java:272]: ohhh 👌

**Pull request merged**